components/settings/SettingsBottomNav.vue

```diff
@@ -0,0 +1,81 @@
+<script setup lang="ts">
+import type { NavButtonName } from '~/composables/settings'
+
+interface NavButton {
+  name: NavButtonName
+  label: string
+  icon: string
+}
+
+const availableNavButtons: NavButton[] = [
+  { name: 'home', label: 'nav.home', icon: 'i-ri:home-5-line' },
+  { name: 'search', label: 'nav.search', icon: 'i-ri:search-line' },
+  { name: 'notification', label: 'nav.notifications', icon: 'i-ri:notification-4-line' },
+  { name: 'mention', label: 'nav.conversations', icon: 'i-ri:at-line' },
+  { name: 'explore', label: 'nav.explore', icon: 'i-ri:compass-3-line' },
+  { name: 'local', label: 'nav.local', icon: 'i-ri:group-2-line' },
+  { name: 'federated', label: 'nav.federated', icon: 'i-ri:earth-line' },
+  // TODO: more menu translatable
+  { name: 'moreMenu', label: '', icon: 'i-ri:more-fill' },
+] as const
+
+const selectedNavButtonNames = ref<NavButtonName[]>(
+  currentUser.value
+    ? ['home', 'search', 'notification', 'mention', 'moreMenu']
+    : ['explore', 'local', 'federated', 'moreMenu'],
+)
+
+const selectedNavButtons = computed<NavButton[]>(() =>
+  selectedNavButtonNames.value.map(name =>
+    availableNavButtons.find(navButton => navButton.name === name)!,
+  ),
+)
+
+function append(navButtonName: NavButtonName) {
+  const maxButtonNumber = 5
+  if (selectedNavButtonNames.value.length < maxButtonNumber)
+    selectedNavButtonNames.value = [...selectedNavButtonNames.value, navButtonName]
+}
+
+function remove() {
+  selectedNavButtonNames.value = selectedNavButtonNames.value.slice(0, selectedNavButtons.value.length - 1)
+}
+</script>
+
+<template>
+  <!-- bottom nav preview -->
+  <div flex="~ gap4 wrap" items-center select-settings h-14 p0>
+    <nav
+      v-for="availableNavButton in selectedNavButtons" :key="availableNavButton.name"
+      flex="~ 1" items-center justify-center text-xl
+      scrollbar-hide overscroll-none
+    >
+      <button btn-base :class="availableNavButton.icon" mx-4 />
+    </nav>
+  </div>
+
+  <nav
+    h-14 flex="~ 1" items-center text-xl
+    scrollbar-hide overscroll-none
+  >
+    <button btn-base i-ri:close-fill text-primary mx-4 @click="remove" />
+  </nav>
+
+  <!-- button selection -->
+  <div flex="~ gap4 wrap" py4>
+    <button
+      v-for="{ name, label, icon } in availableNavButtons"
+      :key="name"
+      btn-text flex="~ gap-2" p2 border="~ base rounded" bg-base ws-nowrap
```

Comment:
maybe you can add `selected` option to `NavButton` to simply logic...

can you add these attributes?
```vue
      type="button"
      role="switch"
      :aria-checked="selectedNavButtonNames.includes(name)"
```

and remove `disable` attribute?

You also need to add remove logic...

Reply:
I'll check it next weekend on my local, don't apply changes yet

Reply:
Thanks for suggestion for button styling! 


I'm actually still explorering what kind of UI would be better overall. A bit more adjustment would be included until finalizing. Maybe I'll add some ordering button or better remove button etc. later for the easy arrangement.

Reply:
Maybe an editable forn? Check push notification in notifications settings (you will needd to enable them, dont forget to remove once you've check it)

Reply:
OK, I was able to implement the toggle button logic now. 🙂